node/src/effect.rs

```diff
@@ -2001,16 +2001,35 @@ impl<REv> EffectBuilder<REv> {
     }
 
     /// Retrieves a `Package` from under the given key in global state if present.
-    pub(crate) async fn get_package(self, state_root_hash: Digest, key: Key) -> Option<Box<Package>>
+    pub(crate) async fn get_package(
+        self,
+        state_root_hash: Digest,
+        package_addr: PackageAddr,
+    ) -> Option<Box<Package>>
     where
         REv: From<ContractRuntimeRequest>,
     {
+        let key = Key::Hash(package_addr);
         let query_request = QueryRequest::new(state_root_hash, key, vec![]);
 
-        if let QueryResult::Success { value, .. } = self.query_global_state(query_request).await {
-            value.into_package().map(Box::new)
-        } else {
-            None
+        match self.query_global_state(query_request).await {
+            QueryResult::RootNotFound | QueryResult::Failure(_) => None,
+            QueryResult::ValueNotFound(_) => {
+                let query_request =
+                    QueryRequest::new(state_root_hash, Key::SmartContract(package_addr), vec![]);
+                error!("requesting under different key");
```

Comment:
this isn't really an error...if we need to log this debug seems like a better fit

Reply:
Will do!

Reply:
I think I added it when the tests were failing must've forgotten to remove it, I think debug makes sense